sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCOptions.scala

```diff
@@ -186,6 +187,11 @@ class JDBCOptions(
 
   // An option to allow/disallow pushing down predicate into JDBC data source
   val pushDownPredicate = parameters.getOrElse(JDBC_PUSHDOWN_PREDICATE, "true").toBoolean
+
+  val connectionFactoryProvider: ConnectionFactoryProvider =
+    parameters.get(JDBC_CONNECTION_FACTORY_PROVIDER).map { className =>
+      Utils.classForName(className).newInstance.asInstanceOf[ConnectionFactoryProvider]
```

Comment:
If the given name is wrong, this seems to end up with `ClassNotFoundException`. Could you add a test case for wrong names and add fallback logic showing a proper warn message and load `DefaultConnectionFactoryProvider` instead?

Reply:
Would you not want to see this exception and know you've done something wrong instead of silently falling back to the default?

Reply:
now throwing IllegalArgumentException with reasonable message, see unit tests